src/commands/TimeCommand.ts

```diff
@@ -0,0 +1,157 @@
+import { Command } from "../Command";
+import { CommandInteraction, GuildMember } from "discord.js";
+import {
+	SlashCommandBooleanOption,
+	SlashCommandBuilder,
+	SlashCommandIntegerOption,
+	SlashCommandStringOption,
+	SlashCommandSubcommandsOnlyBuilder,
+} from "@discordjs/builders";
+
+class Reminder extends Command {
+	constructor() {
+		super("reminder");
+	}
+
+	elapse(interaction: CommandInteraction, callAll: boolean, caller: string, message = ""): void {
+		if (callAll) {
+			interaction.followUp("@everyone, Time's up!");
+		} else {
+			interaction.followUp("<@!" + caller + "> Time's up!");
+		}
+		if (message != "") {
+			interaction.followUp("Message: " + message);
+		}
+		this.m_id -= 1;
+	}
+
+	private rmdDatabase: {
+		id: number;
+		timeout: NodeJS.Timeout;
+		destination: number;
+		message: string;
+		active: boolean;
+	}[] = [];
+	private m_id = 0;
+
+	async execute(interaction: CommandInteraction): Promise<void> {
+		const subcommand = interaction.options.getSubcommand();
+		switch (subcommand) {
+			case "set": {
+				//variables
+				const second = interaction.options.getInteger("second") || 0;
+				const minute = interaction.options.getInteger("minute") || 0;
+				const hour = interaction.options.getInteger("hour") || 0;
+				const message = interaction.options.getString("message") || "";
+				const callAll = interaction.options.getBoolean("callall") || false;
+				const millisecond = (second + 60 * minute + 60 * 60 * hour) * 1000;
+				const d = Date.now() + millisecond;
+				const member = interaction.member as GuildMember;
+				if (callAll) {
+					if (!member.permissions.has("ADMINISTRATOR") && !member.permissions.has("MENTION_EVERYONE")) {
+						await interaction.reply("You don't have the permission, sorry~");
+						break;
+					}
+				}
+				//call setTimeout
+				this.rmdDatabase[this.m_id] = {
+					id: this.m_id,
+					timeout: setTimeout(() => this.elapse(interaction, callAll, interaction.user.id, message), millisecond),
+					destination: d,
+					message: message,
+					active: true,
+				};
+				await interaction.reply(
+					"Okay, I'll remind you soon~\nIn the event that you wish to no longer be reminded of this timer, use /reminder " +
+						this.rmdDatabase[this.m_id].id
+				);
+				this.m_id += 1;
+				break;
+			}
+			case "delete": {
+				//variable
+				const c_id = interaction.options.getInteger("id") || this.m_id - 1;
+				//clear the reminder
+				if (c_id >= this.m_id || c_id < 0) {
+					await interaction.reply("The id does not exist.");
+				} else {
+					clearTimeout(this.rmdDatabase[c_id].timeout);
+					await interaction.reply("I've removed the reminder :))");
+					this.rmdDatabase[c_id].active = false;
```

Comment:
can be removed completely. there is no need to save old reminders and waste storage space

Reply:
I did that because it'd be faster if the id matches the index of the list, so that I won't have to run a for .. loop to find the id needed.

But I'll change it now
=

Reply:
when you use enmap you have a map of id -> reminder so you can delete stuff without any problems